### PR TITLE
Close some gaps in clientset tests

### PR DESCRIPTION
**Pull request by pmorie**
Tests status resources for `Instance` and `Binding`, test update for `ServiceClass`.

cc @jpeeler @MHBauer

### CODE REVIEW

File: test/integration/clientset_test.go [hunk 1]
```diff
@@ -84,77 +85,81 @@ func TestGroupVersion(t *testing.T) {
 // TestNoName checks that all creates fail for objects that have no
 // name given.
 func TestNoName(t *testing.T) {
-	client, shutdownServer := getFreshApiserverAndClient(t)
+	clientset, shutdownServer := getFreshApiserverAndClient(t)
 	defer shutdownServer()
-	scClient := client.Servicecatalog()
+	client := clientset.Servicecatalog()
```
Comment (MHBauer) [inline test/integration/clientset_test.go]: This seems overly pedantic.

File: test/integration/clientset_test.go [hunk 2]
```diff
@@ -164,11 +169,11 @@ func TestBrokerClient(t *testing.T) {
 
 	brokerServer.Spec.AuthSecret = authSecret
 
-	brokerUpdated, err := brokerClient.Update(brokerServer)
+	updated, err := client.Update(brokerServer)
```
Comment (MHBauer) [inline test/integration/clientset_test.go]: I favor leaving broker in the variable name. To be explicit. Is there a refactor you intend to land on top of this that will test all resources with one script? If not, I do not understand why we would change this to be vague.

File: test/integration/clientset_test.go [hunk 3]
```diff
@@ -84,77 +85,81 @@ func TestGroupVersion(t *testing.T) {
...
 		t.Fatal("needs a name", bi.Name)
 	}
 }
 
+// TestBrokerClient exercises the Broker client.
 func TestBrokerClient(t *testing.T) {
-	client, shutdownServer := getFreshApiserverAndClient(t)
+	var (
+		clientset, shutdownServer = getFreshApiserverAndClient(t)
+		client                    = clientset.Servicecatalog().Brokers()
```
Comment (MHBauer) [inline test/integration/clientset_test.go]: This is specifically a client to talk to brokers. Why change the name?

Comment (pmorie) [inline test/integration/clientset_test.go]: I changed the names throughout because the longer names are about 5x harder to read for my eyes.

File: test/integration/clientset_test.go (hunk 1 above)
Comment (pmorie) [inline test/integration/clientset_test.go]: It frees up the 'client' name for use throughout the test.

**Comment by pmorie** [comment]
> I vehemently oppose much of the name changes.

I urge you to conserve energy for things that matter more than variable names.

File: test/integration/clientset_test.go (hunk 2 above)
Comment (pmorie) [inline test/integration/clientset_test.go]: As I said above, a goal was to make this test easier on the eyes to read.  brokerClient isn't super bad, but serviceClassClient crosses some threshold where it just becomes absolutely horrible for my eyes to read (I normally wear contacts, but my eyes are pretty terrible).

I also think we will accumulate enough corner-cases to make it so that we'll want to split out the tests for each resource into their own files, at which point the qualification of names with the type of the resource becomes redundant.

**Comment by pmorie** [comment]
@MHBauer I removed the rename changes from this PR.

File: test/integration/clientset_test.go [hunk 4]
```diff
@@ -104,18 +105,22 @@ func TestNoName(t *testing.T) {
 	}
 }
 
+// TestBrokerClient exercises the Broker client.
 func TestBrokerClient(t *testing.T) {
-	client, shutdownServer := getFreshApiserverAndClient(t)
+	var (
+		client, shutdownServer = getFreshApiserverAndClient(t)
+		brokerClient           = client.Servicecatalog().Brokers()
+		name                   = "test-broker"
```
Comment (arschles) [inline test/integration/clientset_test.go]: can you make this a `const`?

File: test/integration/clientset_test.go [hunk 5]
```diff
@@ -142,19 +147,19 @@ func TestBrokerClient(t *testing.T) {
...
+		t.Fatalf("error getting broker %s (%s)", name, err)
 	}
-	if broker.Name != brokerServer.Name &&
+	if name != brokerServer.Name &&
 		broker.ResourceVersion == brokerServer.ResourceVersion {
 		t.Fatalf("didn't get the same broker back from the server \n%+v\n%+v", broker, brokerServer)
 	}
 
-	// check that the broker is the same both ways
+	// check that the broker is the from get and list
```
Comment (arschles) [inline test/integration/clientset_test.go]: add a "same" between "the" and "from get and list"

File: test/integration/clientset_test.go [hunk 6]
```diff
@@ -265,51 +274,67 @@ func TestServiceClassClient(t *testing.T) {
...
 		t.Fatalf("error listing service classes (%s)", err)
 	}
-	if serviceClassAtServer.Name != serviceClass.Name &&
+	if serviceClassAtServer.Name != name &&
 		serviceClass.ResourceVersion == serviceClassAtServer.ResourceVersion {
 		t.Fatalf("didn't get the same ServiceClass back from the server \n%+v\n%+v", serviceClass, serviceClassAtServer)
 	}
 
-	err = serviceClassClient.Delete("test-serviceclass", &v1.DeleteOptions{})
+	// check that the broker is the from get and list
```
Comment (arschles) [inline test/integration/clientset_test.go]: same here RE "same"

File: test/integration/clientset_test.go [hunk 7]
```diff
@@ -265,51 +274,67 @@ func TestServiceClassClient(t *testing.T) {
...
-				{
-					Type:    v1alpha1.InstanceConditionReady,
-					Status:  v1alpha1.ConditionTrue,
-					Reason:  "reason",
-					Message: "message",
-				},
+	var (
+		client, shutdownServer = getFreshApiserverAndClient(t)
+		instanceClient         = client.Servicecatalog().Instances("test-namespace")
+		name                   = "test-instance"
```
Comment (arschles) [inline test/integration/clientset_test.go]: same here RE `const name`

File: test/integration/clientset_test.go [hunk 8]
```diff
@@ -362,51 +392,64 @@ func TestInstanceClient(t *testing.T) {
...
-			Conditions: []v1alpha1.BindingCondition{
-				{
-					Type:    v1alpha1.BindingConditionReady,
-					Status:  v1alpha1.ConditionTrue,
-					Reason:  "reason",
-					Message: "message",
+	var (
+		client, shutdownServer = getFreshApiserverAndClient(t)
+		bindingClient          = client.Servicecatalog().Bindings("test-namespace")
+		name                   = "test-binding"
```
Comment (arschles) [inline test/integration/clientset_test.go]: same here RE `const name`

File: test/integration/clientset_test.go [hunk 9]
```diff
@@ -469,13 +517,34 @@ func TestBindingClient(t *testing.T) {
...
+		Type:    v1alpha1.BindingConditionReady,
+		Status:  v1alpha1.ConditionTrue,
+		Reason:  "ConditionReason",
+		Message: "ConditionMessage",
+	}
+	bindingServer.Status = v1alpha1.BindingStatus{
+		Conditions: []v1alpha1.BindingCondition{readyConditionTrue},
+	}
+	_, err = bindingClient.UpdateStatus(bindingServer)
+	if err != nil {
```
Comment (arschles) [inline test/integration/clientset_test.go]: can you combine this and the previous line:

```go
if _, err := bindingClient.UpdateStatus(bindingServer); err != nil {
```

File: test/integration/clientset_test.go [hunk 10]
```diff
@@ -469,13 +517,34 @@ func TestBindingClient(t *testing.T) {
...
+	bindingServer, err = bindingClient.Get(name)
+	if err != nil {
+		t.Fatalf("error getting binding: %v", err)
+	}
+	if e, a := readyConditionTrue, bindingServer.Status.Conditions[0]; !reflect.DeepEqual(e, a) {
+		t.Fatal("Didn't get matching ready conditions:\nexpected: %v\n\ngot: %v", e, a)
+	}
+
+	err = bindingClient.Delete(name, &v1.DeleteOptions{})
 	if nil != err {
```
Comment (arschles) [inline test/integration/clientset_test.go]: same here RE combining this and the previous line

File: test/integration/clientset_test.go (hunk 4 above)
Comment (pmorie) [inline test/integration/clientset_test.go]: done in next push

File: test/integration/clientset_test.go (hunk 5 above)
Comment (pmorie) [inline test/integration/clientset_test.go]: done in next push

File: test/integration/clientset_test.go (hunk 6 above)
Comment (pmorie) [inline test/integration/clientset_test.go]: done

File: test/integration/clientset_test.go (hunk 7 above)
Comment (pmorie) [inline test/integration/clientset_test.go]: done

File: test/integration/clientset_test.go (hunk 8 above)
Comment (pmorie) [inline test/integration/clientset_test.go]: done

**Comment by arschles** [comment]
I'm merging this now so we can move forward with testing before MVP1 is due
